test/integration/directory_artifact_test.js

```diff
@@ -79,4 +83,81 @@ suite('Directory artifact', () => {
       'Size of uploaded contents of test.html does not match original.'
     );
   });
+
+  test('retry directory upload', async () => {
+    let ARTIFACT_COUNT = 3;
+
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    var count = 0;
+
+    let rejectConnection = function() {
+      // Delay iptables rejection to catch
+      // in the middle of the upload
+      setTimeout(function() {
+        iptables.reject({
```

Comment:
Hrm, could we make this test disabled by default unless running in automation or something? I just do not want someone running this locally screwing up their ip tables on their host.